internal/service/vpc/resource_gcp_vpc_peering_connection_test.go

```diff
@@ -0,0 +1,204 @@
+package vpc_test
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/aiven/aiven-go-client"
+	acc "github.com/aiven/terraform-provider-aiven/internal/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/kelseyhightower/envconfig"
+)
+
+type gcpSecrets struct {
+	GCPProjectID string `envconfig:"GCP_PROJECT_ID" required:"true"`
+	GCPRegion    string `envconfig:"GCP_REGION" required:"true"`
+}
+
+func TestAccAivenGCPPeeringConnection_basic(t *testing.T) {
+	var s gcpSecrets
+
+	err := envconfig.Process("", &s)
+	if err != nil {
+		t.Skipf("Not all values have been provided to establish a GCP VPC peering connection: %s", err)
+	}
+
+	importResourceName := "aiven_gcp_vpc_peering_connection.foo"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { acc.TestAccPreCheck(t) },
+		ProviderFactories: acc.TestAccProviderFactories,
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"google": {
+				Source:            "hashicorp/google",
+				VersionConstraint: ">=4.43.1",
+			},
+		},
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGCPVPCPeeringConnection(&s),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(importResourceName, "state"),
+					resource.TestCheckResourceAttrSet(importResourceName, "self_link"),
+					resource.TestCheckResourceAttr("google_compute_network_peering.foo", "state", "ACTIVE"),
+					resource.TestCheckResourceAttr("data.aiven_gcp_vpc_peering_connection.bar", "state", "PENDING_PEER"),
+					resource.TestCheckResourceAttrSet("data.aiven_gcp_vpc_peering_connection.bar", "self_link"),
+				),
+			},
+			{
+				Config: testAccGCPVPCPeeringConnection(&s),
+				Check: func(state *terraform.State) error {
+					c := acc.TestAccProvider.Meta().(*aiven.Client)
+					p := os.Getenv("AIVEN_PROJECT_NAME")
+
+				QueryVpc:
+					vpcs, err := c.VPCs.List(p)
+					if err != nil {
+						return err
+					}
+
+					var v *aiven.VPC
+					for _, vpc := range vpcs {
+						if vpc.CloudName == fmt.Sprintf("google-%s", os.Getenv("GCP_REGION")) {
+							v = vpc
+						}
+					}
+
+					if v == nil {
+						return errors.New("error getting GCP peering connection, project VPC is empty")
+					}
+
+					cons, err := c.VPCPeeringConnections.List(p, v.ProjectVPCID)
+					if err != nil {
+						return fmt.Errorf("error getting list of peering connections: %w", err)
+					}
+
+					if len(cons) == 0 {
+						return fmt.Errorf("error getting GCP peering connection, list of peering connections is empty for Project VPC ID (%s)", v.ProjectVPCID)
+					}
+
+					for _, pvpc := range cons {
+						if pvpc.State != "ACTIVE" {
```

Comment:
we should have an enum-like list of consts and store `ACTIVE` and other possible values in it

Reply:
We do not have anything like that anywhere. We do not need to refactor this part of the code for all VPC peering resources within this PR. Sounds like a separate task.